### PR TITLE
Implemented part of solving quiz UI

### PR DESCRIPTION
**Pull request by marinmarkelic**
https://user-images.githubusercontent.com/101728747/181497887-d82ba4b6-8c34-4896-86d3-9dc1e629a45f.mp4

Still need to add:
- Changing color of correct and incorrect answers
- Error message if an error occurs
- Invalidating collectionview layout on rotation
- View with a score when the quiz is finished

### CODE REVIEW

File: QuizApp/QuizApp/UI/Screens/Quiz/QuizModels.swift [hunk 1]
```diff
@@ -139,6 +139,14 @@ struct QuizStartResponse {
 
 }
 
+extension QuizStartResponse {
+
+    static var empty: QuizStartResponse = {
```
Comment (istaudacher) [inline QuizApp/QuizApp/UI/Screens/Quiz/QuizModels.swift]: Why not `static let empty = QuizStartResponse(questions: [], sessionId: "")`.

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/AnswerView.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import UIKit
+
+class AnswerView: UIView {
+
+    weak var delegate: AnswerViewDelegate!
```
Comment (istaudacher) [inline QuizApp/QuizApp/UI/Screens/SolvingQuiz/AnswerView.swift]: `weak var delegate: AnswerViewDelegate?`

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/AnswerView.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        addGestureRecognizer(gesture)
+
+        label.font = UIFont(name: "SourceSansPro-Bold", size: 20)
+        label.textColor = .white
+        label.numberOfLines = 0
+    }
+
+    @objc
+    private func selectedAnswer() {
+        delegate.selectedAnswer(with: id ?? 0)
```
Comment (istaudacher) [inline QuizApp/QuizApp/UI/Screens/SolvingQuiz/AnswerView.swift]: If the id is not set, that means something has gone wrong in the development. Why is there a default value, it is not declared as an optional?

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/ProgressView.swift [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func set(colors: [UIColor]) {
+        stackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
+
+        for index in 0..<colors.count {
+            if colors[index] == UIColor.white {
+                label.text = "\(index + 1)/\(colors.count)"
```
Comment (istaudacher) [inline QuizApp/QuizApp/UI/Screens/SolvingQuiz/ProgressView.swift:25]: You are making a decision based on color. That is not correct. Instead, pass some variables to indicate what is the current questing index and what is the total number of questions.

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionCell.swift [hunk 5]
```diff
@@ -0,0 +1,86 @@
+import UIKit
+
+class QuestionCell: UICollectionViewCell {
+
+    static let reuseIdentifier = String(describing: QuestionCell.self)
+
+    weak var delegate: QuestionCellDelegate!
```
Comment (istaudacher) [inline QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionCell.swift]: `weak var delegate: QuestionCellDelegate?`

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionCell.swift [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    func defineLayoutForViews() {
+        label.snp.makeConstraints {
+            $0.leading.top.trailing.equalToSuperview()
+        }
+
+        stackView.snp.makeConstraints {
+            $0.top.equalTo(label.snp.bottom).offset(30)
+            $0.leading.trailing.equalToSuperview()
```
Comment (istaudacher) [inline QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionCell.swift:68]: Where is bottom?

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionsView.swift [hunk 7]
```diff
@@ -0,0 +1,134 @@
+import UIKit
+
+class QuestionsView: UIView {
+
+    weak var delegate: QuestionsViewDelegate!
```
Comment (istaudacher) [inline QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionsView.swift]: `weak var delegate: QuestionsViewDelegate?`

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionsView.swift [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func set(questions: [Question]) {
+        self.questions = questions
+        collectionView.reloadData()
+    }
+
+    func scrollToQuestion(at index: Int?) {
```
Comment (istaudacher) [inline QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionsView.swift:29]: Should you be able to scroll to a question if you do not know where to scroll? In other words, should the index be optional?

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/SolvingQuizViewModel.swift [hunk 9]
```diff
@@ -1,24 +1,56 @@
...
+                progressColors[0] = .white
             } catch {
 
             }
         }
     }
 
+    @MainActor
+    func selectedAnswer(with id: Int) {
+        guard let questionIndex = progressColors.firstIndex(of: .white) else { return }
```
Comment (istaudacher) [inline QuizApp/QuizApp/UI/Screens/SolvingQuiz/SolvingQuizViewModel.swift:38]: Please do not make conclusions based on colors. There must be a way to track the current question.

**Review by istaudacher — CHANGES_REQUESTED** [review]

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/SolvingQuizViewModel.swift (hunk 9 above)
Comment (marinmarkelic) [reply QuizApp/QuizApp/UI/Screens/SolvingQuiz/SolvingQuizViewModel.swift:38]: Solved on next branch

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/ProgressView.swift (hunk 4 above)
Comment (marinmarkelic) [reply QuizApp/QuizApp/UI/Screens/SolvingQuiz/ProgressView.swift:25]: Solved on next branch

File: QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionsView.swift (hunk 8 above)
Comment (marinmarkelic) [reply QuizApp/QuizApp/UI/Screens/SolvingQuiz/QuestionsView.swift:29]: Solved on next branch

**Review by istaudacher — APPROVED** [review]

**Pull request merged**